src/controllers/AnnotationModeController.js

```diff
@@ -219,36 +225,24 @@ class AnnotationModeController extends EventEmitter {
      * @return {void}
      */
     unregisterThread(thread) {
-        const page = thread.location.page || 1; // Defaults to page 1 if thread has no page'
-        const pageThreads = this.threads[page] || {};
+        if (!thread || !thread.location) {
+            return;
+        }
 
-        delete pageThreads[thread.threadID];
-        this.threads[page] = pageThreads;
+        const page = thread.location.page || 1; // Defaults to page 1 if thread has no page'
+        this.threads[page].remove(thread);
```

Comment:
Is it possible that `this.threads[1]` does not contain the thread (in the case of `thread.location.page` being null)? If so, how does `.remove()` behave?

Reply:
rbush would only attempt to do something if the thread was found in the tree https://github.com/mourner/rbush/blob/master/index.js#L152